aws_eden_cli/dynamodb.py

```diff
@@ -167,11 +178,62 @@ def fetch_all_environments(table):
     return environments
 
 
+def fetch_all_profiles(table):
+    profiles = {}
+
+    try:
+        r = table.query(
+            KeyConditionExpression=Key('type').eq('_profile')
+        )
+    except Exception as e:
+        if hasattr(e, 'response') and 'Error' in e.response:
+            logger.error(e.response['Error']['Message'])
+            return None
+        else:
+            logger.error(f"Unknown exception raised: {e}")
+            return None
+
+    for item in r['Items']:
+        logger.info(item)
+        profiles += {
+            'name': item['name'],
+            'profile': item['profile'],
+        }
+
+    return profiles
+
+
+def fetch_profile(table, profile_name):
+    try:
+        r = table.query(
+            KeyConditionExpression=Key('type').eq('_profile') & Key('name').eq(profile_name)
+        )
+    except Exception as e:
+        if hasattr(e, 'response') and 'Error' in e.response:
+            logger.error(e.response['Error']['Message'])
+            return None
+        else:
+            logger.error(f"Unknown exception raised: {e}")
+            return None
+
+    if 'Items' not in r or len(r['Items']) == 0:
+        logger.warning(f"Profile {profile_name} not found in remote table!")
+        return None
+    elif len(r['Items']) != 1:
+        logger.warning(f"Got multiple profiles for profile name {profile_name} from remote table!")
+        return None
```

Comment:
How about returning the profile with first or last match in remote tables?

Reply:
Order on multiple items is not guaranteed AFAIK, thus I think we should fail here.

Reply:
This should not be normally happening. I wanted to see when this happens in wild actually.